spec/support/helpers/url_helper.rb

```diff
@@ -5,7 +5,19 @@ def token_endpoint_url(options = {})
       client_id: options[:client_id]     || (options[:client] ? options[:client].uid : nil),
       client_secret: options[:client_secret] || (options[:client] ? options[:client].secret : nil),
       redirect_uri: options[:redirect_uri]  || (options[:client] ? options[:client].redirect_uri : nil),
-      grant_type: options[:grant_type]    || 'authorization_code'
+      grant_type: options[:grant_type]    || 'authorization_code',
+      code_challenge_method: 'plain'
```

Comment:
@rsharrott You don't need to send this up on the token exchange. The code shouldn't (and I don't think it does) care about this param on the token exchange because this method was established on the authorization request.

Reply:
Also, I made a separate method for the PKCE related stuff, `pkce_token_endpoint_url`.